boa_engine/src/object/jstypedarray.rs

```diff
@@ -0,0 +1,407 @@
+use crate::{
+    builtins::typed_array::TypedArray,
+    object::{JsArray, JsObject, JsObjectType},
+    value::IntoOrUndefined,
+    Context, JsResult, JsString, JsValue,
+};
+use boa_gc::{Finalize, Trace};
+use std::ops::Deref;
+
+/// JavaScript `TypedArray` rust object.
+#[derive(Debug, Clone, Trace, Finalize)]
+pub struct JsTypedArray {
+    inner: JsObject,
```

Comment:
Does it make sense to have `inner` as a `JsObject`? Almost every usage of `inner` has to clone the gc pointer, wrap it into a `JsValue` and pass that reference.

Reply:
> Does it make sense to have `inner` as a `JsObject`? Almost every usage of `inner` has to clone the gc pointer, wrap it into a `JsValue` and pass that reference.

Hmmm.... Semantically it should be an object, but from a performance point of view it should be a `JsValue` so we don't clone and wrap it.

In all but one case (where we call `Deref<JsObject>`) we need it as `JsValue`